pyteal/compiler/compiler.py

```diff
@@ -167,6 +172,7 @@ def compileTeal(
     *,
     version: int = DEFAULT_TEAL_VERSION,
     assembleConstants: bool = False,
+    optimize: OptimizeOptions = OptimizeOptions(),
```

Comment:
Good point, I didn't know this about python. I'll change it to None.

Reply:
Leaving as a reference, not a change request:  It _is_ possible to make the object immutable because it's a `dataclass`.  From https://docs.python.org/3/library/dataclasses.html:

> frozen: If true (the default is False), assigning to fields will generate an exception. This emulates read-only frozen instances. If __setattr__() or __delattr__() is defined in the class, then [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) is raised. See the discussion below.